src/ResourceManager/DeviceProvisioningServices/Commands.DeviceProvisioningServices/Cmdlet/IotDpsCertificates/SetAzureRmIotDeviceProvisioningServiceCertificate.cs

```diff
@@ -0,0 +1,178 @@
+        public string CertificateName { get; set; }
+
+        [Parameter(
+           Position = 1,
+           Mandatory = true,
+           ParameterSetName = ResourceIdParameterSet,
+           HelpMessage = "Etag of the Certificate")]
+        [Parameter(
+           Position = 3,
+           Mandatory = true,
+           ParameterSetName = ResourceParameterSet,
+           HelpMessage = "Etag of the Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string Etag { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            HelpMessage = "base-64 representation of X509 certificate .cer file or .pem file path")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "base-64 representation of X509 certificate .cer file or .pem file path")]
+        [Parameter(
+            Position = 4,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "base-64 representation of X509 certificate .cer file or .pem file path")]
+        [ValidateNotNullOrEmpty]
+        public string Path { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Name, DPSResources.SetCertificate))
+            {
+                switch (ParameterSetName)
+                {
+                    case InputObjectParameterSet:
+                        this.ResourceGroupName = this.InputObject.ResourceGroupName;
+                        this.Name = this.InputObject.Name;
+                        this.CertificateName = this.InputObject.CertificateName;
+                        this.Etag = this.InputObject.Etag;
+                        this.SetIotDpsCertificate();
+                        break;
+
+                    case ResourceIdParameterSet:
+                        this.ResourceGroupName = IotDpsUtils.GetResourceGroupName(this.ResourceId);
+                        this.Name = IotDpsUtils.GetIotDpsName(this.ResourceId);
+                        this.CertificateName = IotDpsUtils.GetIotDpsCertificateName(this.ResourceId);
+                        this.SetIotDpsCertificate();
+                        break;
+
+                    case ResourceParameterSet:
+                        this.SetIotDpsCertificate();
+                        break;
+
+                    default:
+                        throw new ArgumentException("BadParameterSetName");
+                }
+            }
+        }
+
+        private void SetIotDpsCertificate()
+        {
+            string certificate = string.Empty;
+            FileInfo fileInfo = new FileInfo(this.Path);
+            switch (fileInfo.Extension.ToLower(CultureInfo.InvariantCulture))
+            {
+                case ".cer":
+                    var certificateByteContent = File.ReadAllBytes(this.Path);
```

Comment:
@anusapan whenever working with the file system, please use `AzureSession.Instance.DataStore` to read from/write to files. This applies to all instances of `File.*`

Reply:
Done